feg/gateway/services/session_proxy/credit_control/gx/model_conversion.go

```diff
@@ -295,20 +295,24 @@ func (report *UsageReport) FromUsageMonitorUpdate(update *protos.UsageMonitorUpd
 }
 
 func (monitor *UsageMonitoringInfo) ToUsageMonitoringCredit() *protos.UsageMonitoringCredit {
-	if monitor.GrantedServiceUnit == nil || monitor.GrantedServiceUnit.IsEmpty() {
-		return &protos.UsageMonitoringCredit{
-			Action:        protos.UsageMonitoringCredit_DISABLE,
-			MonitoringKey: monitor.MonitoringKey,
-			Level:         protos.MonitoringLevel(monitor.Level),
-		}
-	} else {
-		return &protos.UsageMonitoringCredit{
-			Action:        protos.UsageMonitoringCredit_CONTINUE,
-			MonitoringKey: monitor.MonitoringKey,
-			GrantedUnits:  monitor.GrantedServiceUnit.ToProto(),
-			Level:         protos.MonitoringLevel(monitor.Level),
-		}
+	return &protos.UsageMonitoringCredit{
+		Action:        monitor.ToUsageMonitoringAction(),
+		MonitoringKey: monitor.MonitoringKey,
+		GrantedUnits:  monitor.GrantedServiceUnit.ToProto(),
+		Level:         protos.MonitoringLevel(monitor.Level),
+	}
+}
+
+func (monitor *UsageMonitoringInfo) ToUsageMonitoringAction() protos.UsageMonitoringCredit_Action {
```

Comment:
Not sure if it exists but can we start adding comments with spec version + section number? Makes it easier later on for debugging.